### ISSUE TITLE
508-defect-3 [COGNITION] Direct Deposit Component - Header should be updated when users switch into edit mode

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va-forms-system-core#317]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I want to hear that I have switched into the edit inline "state" of the direct deposit component.

## Details

Screen readers do not have any aural cues that users have entered or exited the edit in place mode. We have a state Boolean `editing` that could be used to set the component header conditionally. Screenshot attached below.

## Acceptance Criteria

- [x] Component header says "Edit direct deposit information" when editing inline
- [x] Component header says "Direct deposit information" when read-only
- [x] Zero axe-core violations

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2021-01-12 at 4 02 24 PM](https://user-images.githubusercontent.com/934879/104379863-b6757500-54ef-11eb-87a9-5803357c102f.png)

**Comment by caw310** [comment]
Reassigning to FLT backlog after talking with Matt and Katy.

**Comment by taylorkaren** [comment]
@obliviga: Note that there are other open accessibility tickets related to the Direct Deposit component. You may want to work on all of these tickets at the same time:
- https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/313
- https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/316
- https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/255

**Comment by obliviga** [comment]
Hi @caw310, could you provide an example of the page URL in which this form is located? It looks like the POC for this ticket no longer works at Ad Hoc. Thank you!

**Comment by caw310** [comment]
@bkjohnson @k80bowman , can you provide example?

**Comment by bkjohnson** [comment]
I don't know which form was used for that specific screenshot, but the `directDeposit` definition from the forms system is [imported here](https://github.com/department-of-veterans-affairs/vets-website/blob/ef6ceb2ab1b966b1f61775e9b3d6ceaed5a8772c/src/applications/edu-benefits/1990s/pages/directDeposit.jsx#L1) and used with some overriden values. You can  find it in the form starting at the [1990s rootUrl](https://staging.va.gov/education/other-va-education-benefits/veteran-rapid-retraining-assistance/apply-for-vrrap-form-22-1990s):

![Screenshot of direct deposit pattern in VRAPP form](https://user-images.githubusercontent.com/2008881/173714933-ee45c12a-6536-4cb2-9625-f86ba99ea36a.png)

Be aware that some applications are not using the definition from the forms library and are instead creating their own:

- [`edu-benefits/10203`](https://github.com/department-of-veterans-affairs/vets-website/blob/ef6ceb2ab1b966b1f61775e9b3d6ceaed5a8772c/src/applications/edu-benefits/10203/pages/directDeposit.js)
- [`edu-benefits/0994`](https://github.com/department-of-veterans-affairs/vets-website/blob/ef6ceb2ab1b966b1f61775e9b3d6ceaed5a8772c/src/applications/edu-benefits/0994/pages/bankInformation.js)
- [`edu-benefits/1990e`](https://github.com/department-of-veterans-affairs/vets-website/blob/ef6ceb2ab1b966b1f61775e9b3d6ceaed5a8772c/src/applications/edu-benefits/1990e/config/form.js#L369-L370)

Also, the personalization app [has some direct deposit functionality](https://github.com/department-of-veterans-affairs/vets-website/blob/ef6ceb2ab1b966b1f61775e9b3d6ceaed5a8772c/src/applications/personalization/profile/components/direct-deposit/DirectDeposit.jsx) but that may be separate from the forms system.

**Comment by obliviga** [comment]
Exactly what I needed, thank you @bkjohnson !

**Comment by obliviga** [comment]
Hi @taylorkaren, can we remove the "Focus is managed properly as noted in #313" acceptance criteria from this ticket? I've found the fix for the main task in this ticket, and can create a PR for it. Then we can close this ticket and I could create a separate PR for #313, since that issue is a bit more complex.

**Comment by taylorkaren** [comment]
@obliviga - Yes. I have removed the - [ ] Focus is managed properly as noted in department-of-veterans-affairs/va-forms-system-core#313 AC from this ticket.

**Comment by obliviga** [comment]
Fix has been verified in production: https://adhoc.slack.com/files/U03ETG3PG6T/F03LQ2GFZJ9/direct_deposit_functionality_prod.mov
